pkg/gofr/datasource/file/sftp/fs_test.go

```diff
@@ -0,0 +1,304 @@
+package sftp
+
+import (
+	"errors"
+	"os"
+	"testing"
+
+	"github.com/pkg/sftp"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+	File "gofr.dev/pkg/gofr/datasource/file"
+)
+
+type mocks struct {
+	client  *MocksftpClient
+	logger  *MockLogger
+	metrics *MockMetrics
+	file    *File.MockFile
+}
+
+func getMocks(t *testing.T) (fileSystem, mocks) {
+	t.Helper()
+
+	ctrl := gomock.NewController(t)
+
+	mockClient := NewMocksftpClient(ctrl)
+	mockLogger := NewMockLogger(ctrl)
+	mockMetrics := NewMockMetrics(ctrl)
+	mockFile := File.NewMockFile(ctrl)
+
+	client := fileSystem{logger: mockLogger, metrics: mockMetrics, client: mockClient}
+
+	return client, mocks{
+		client:  mockClient,
+		logger:  mockLogger,
+		metrics: mockMetrics,
+		file:    mockFile,
+	}
+}
+
+func TestFiles_Mkdir(t *testing.T) {
+	files, mocks := getMocks(t)
+
+	testCases := []struct {
+		desc string
+		err  error
+	}{
+		{"directory created successfully", nil},
+		{"directory creation failed", errors.New("failed to create file")},
+	}
+
+	for _, tc := range testCases {
+		mocks.client.EXPECT().Mkdir("/test").Return(tc.err)
+
+		err := files.Mkdir("/test", 1)
+
+		require.Equal(t, tc.err, err, "TEST[%d] Failed. Desc %v")
```

Comment:
I would have used `require.ErrorContains`  and do not define local errors with errors.New, or maybe `require.ErrorEqual`

This comment is valid in various places in this test file

Reply:
I have multiple cases where some cases also have nil error and ErrorEqual expects an error and a string - I would need to check empty string with error? should that be better compared to what I am doing now? @ccoVeille

Reply:
I would say you could use a string in the struct, then when the variable is empty you use require.NoError, then when the string is not empty, you can use ErrorContains or EqualError

Reply:
I am not fully convinced with this.
I believe it's better to check the error as nil in case of no error - there could be a mistake by the person who is writing the code and return an empty string - in that case also it will pass.